compiler/next/include/chpl/uast/Function.h

```diff
@@ -0,0 +1,328 @@
+   */
+  int numFormals() const {
+    return numFormals_; 
+  }
+
+  /**
+   Return the i'th formal
+   */
+  const Formal* formal(int i) const {
+    assert(numFormals_ > 0 && formalsChildNum_ >= 0);
+    assert(0 <= i && i < numFormals_);
+    const ASTNode* ast = this->child(formalsChildNum_ + i);
+    assert(ast->isFormalDecl());
+    const FormalDecl* d = (const FormalDecl*) ast;
+    return d->formal();
+  }
+
+  const Formal* thisFormal() const {
+    if (thisFormalChildNum_ >= 0) {
+      const ASTNode* ast = this->child(thisFormalChildNum_);
+      assert(ast->isFormalDecl());
+      const FormalDecl* d = (const FormalDecl*) ast;
+      return d->formal();
+    } else {
+      return nullptr;
+    }
+  }
+
+  /**
+   Returns the expression for the return type or nullptr if there was none.
+   */
+  const Expression* returnType() const {
+    if (returnTypeChildNum_ >= 0) {
+      const ASTNode* ast = this->child(returnTypeChildNum_);
+      assert(ast->isExpression());
+      return (Expression*) ast; 
+    } else {
+      return nullptr;
+    }
+  }
+
+  /**
+   Returns the expression for the where clause or nullptr if there was none.
+   */
+  const Expression* whereClause() const {
+    if (whereChildNum_ >= 0) {
+      const ASTNode* ast = this->child(whereChildNum_);
+      assert(ast->isExpression());
+      return (const Expression*) ast;
+    } else {
+      return nullptr;
+    }
+  }
+
+  /**
+   Return a way to iterate over the lifetime clauses.
+   */
+  ASTListIteratorPair<Expression> lifetimeClauses() const {
+    if (numLifetimeParts_ == 0) {
+      return ASTListIteratorPair<Expression>(children_.end(), children_.end());
+    } else {
+      auto start = children_.begin() + lifetimeChildNum_;
+      return ASTListIteratorPair<Expression>(start, start + numLifetimeParts_);
+    }
+  }
+
+  /**
+   Return the number of lifetime clauses
+   */
+  int numLifetimeClauses() const {
+    return numLifetimeParts_;
+  }
+
+  /**
+   Return the i'th lifetime clause
+   */
+  const Expression* lifetimeClause(int i) const {
+    assert(numLifetimeParts_ > 0 && lifetimeChildNum_ >= 0);
+    assert(0 <= i && i < numLifetimeParts_);
+    const ASTNode* ast = this->child(lifetimeChildNum_ + i);
+    assert(ast->isExpression());
+    return (const Expression*) ast;
+  }
+
+  /**
+    Return a way to iterate over the statements in the function body.
+   */
+  ASTListIteratorPair<Expression> stmts() const {
+    if (numBodyStmts_ == 0) {
+      return ASTListIteratorPair<Expression>(children_.end(), children_.end());
+    } else {
+      auto start = children_.begin() + bodyChildNum_;
+      return ASTListIteratorPair<Expression>(start, start + numBodyStmts_);
+    }
+  }
+
+  /**
+   Return the number of statements in the function body.
+   */
+  int numStmts() const {
+    return numBodyStmts_;
+  }
+
+  /**
+   Return the i'th statement in the function body.
+   */
+  const Expression* stmt(int i) const {
```

Comment:
I'm interested for suggestions for the name of this function. Is `stmt` clear enough? Should it mention `body` ? E.g. `bodyStmt(int i)` ? Or is there something better?

Reply:
See https://github.com/Cray/chapel-private/issues/1978